src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -811,6 +819,32 @@ public void RecordProperties_10()
                 );
         }
 
+        [Fact, WorkItem(48584, "https://github.com/dotnet/roslyn/issues/48584")]
+        public void RecordProperties_11_UnusedPositionalParameter()
+        {
+            var comp = CreateCompilation(@"
+record C1(object O1, object O2, object O3)
+{
+    public object O1 { get; init; }
+    public object O2 { get; init; } = M(O2);
+    public object O3 { get; init; } = M(O3 = null);
+    private static object M(object o) => o;
+}
+record C2(object O4) : C1(O4, O4, O4) // we didn't complain because the parameter is read
+{
+    public object O4 { get; init; }
+}
+");
+            comp.VerifyDiagnostics(
+                    // (2,18): warning CS8907: Parameter 'O1' is unread. Did you forget to use it to initialize the property with that name?
+                    // record C1(object O1, object O2, object O3)
+                    Diagnostic(ErrorCode.WRN_UnusedRecordParameter, "O1").WithArguments("O1").WithLocation(2, 18),
+                    // (2,40): warning CS8907: Parameter 'O3' is unread. Did you forget to use it to initialize the property with that name?
```

Comment:
>warning CS8907: Parameter 'O3' is unread. Did you forget to use it to initialize the property with that name? [](start = 31, length = 109)

This warning feels unexpected to me. The parameter is used.

Reply:
I should have updated the PR title, but didn't want to mess email threading. The main intent for this warning is that people are surprised when they write `record C(int X) { public int X { get; init; } }` and the property is not initialized. So I think it makes sense to focus on reads.

If one does `record C(int X) : Base(X) { public int X { get; init; } };`, I think a warning would also be desirable for that reason, but I can see that's debatable.

Started a thread with LDM to clarify expectations.

---
In reply to: [512822355](https://github.com/dotnet/roslyn/pull/48895#discussion_r512822355) [](ancestors = 512822355)

Reply:
Clarified with LDM. We'll warn on *unread* parameters. Didn't get feedback on phrasing of warning, but I'm open to proposal (let me know).

---
In reply to: [512891377](https://github.com/dotnet/roslyn/pull/48895#discussion_r512891377) [](ancestors = 512891377,512822355)